src/tallies/filter_energy.cpp

```diff
@@ -18,6 +19,7 @@ namespace openmc {
 void EnergyFilter::from_xml(pugi::xml_node node)
 {
   auto bins = get_node_array<double>(node, "bins");
+  simulation::bins_pht = bins; //! store them as a global variable for the pht
```

Comment:
This is not a good design. If you have multiple energy filters in your problem, each of them will be trying to modify global state. it wont' work if anyone tries to tally any other energy-dependent quantities.

Reply:
Yes, I will also revise this part completely and think about how to pass the `pht_storage` of the particle to the tally and the filters.

Reply:
Agree with @gridley. I'd encourage you to think about the user interface for this feature a bit since that will influence how data is stored on the C++ side. For example, would the following be appropriate?
```Python
ph_tally = openmc.Tally()
ph_tally.filters = [openmc.CellFilter(...), openmc.EnergyFilter(...)]
ph_tally.scores = ['pulse-height']
```
It may be the case that pulse-height tallies are different enough that they deserve their own class, in which case it might look something like:
```Python
ph_tally = openmc.PulseHeightTally()
ph_tally.filters = [openmc.CellFilter(...), openmc.EnergyFilter(...)]
```
I'm not sure offhand whether this latter idea is necessary. We may have to think a bit more carefully about the overall design of different tally types.